torchreid/integration/sc/train_task.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import logging
+import math
+import os
+import re
+from copy import deepcopy
+from typing import List, Optional
+
+import torchreid
+from ote_sdk.entities.datasets import DatasetEntity
+from ote_sdk.entities.metrics import (CurveMetric, LineChartInfo, LineMetricsGroup,
+                                      MetricsGroup, Performance, ScoreMetric)
+from ote_sdk.entities.model import ModelEntity, ModelStatus
+from ote_sdk.entities.subset import Subset
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.entities.train_parameters import default_progress_callback, TrainParameters
+from ote_sdk.usecases.tasks.interfaces.training_interface import ITrainingTask
+from scripts.default_config import imagedata_kwargs, lr_scheduler_kwargs, optimizer_kwargs
+from torchreid.apis.training import run_lr_finder, run_training
+from torchreid.integration.sc.inference_task import OTEClassificationInferenceTask
+from torchreid.integration.sc.monitors import DefaultMetricsMonitor
+from torchreid.integration.sc.utils import (OTEClassificationDataset, save_model,
+                                            TrainingProgressCallback)
+from torchreid.ops import DataParallel
+from torchreid.utils import load_checkpoint, load_pretrained_weights, set_random_seed
+
+logger = logging.getLogger(__name__)
+
+
+class OTEClassificationTrainingTask(OTEClassificationInferenceTask, ITrainingTask):
+
+    def __init__(self, task_environment: TaskEnvironment):
+        super().__init__(task_environment)
+        self._aux_model_snap_paths = {}
+
+    def cancel_training(self):
+        """
+        Called when the user wants to abort training.
+        In this example, this is not implemented.
+
+        :return: None
+        """
+        logger.info("Cancel training requested.")
+        self.stop_callback.stop()
+
+    def save_model(self, output_model: ModelEntity):
```

Comment:
The save_model function is part of the ITrainingTask interface and should be implemented in the OTEClassificationTrainingTask. I can create the common _save_model function in the OTEClassificationInferenceTask. What do you think?

Reply:
It is not critical. Just not very pretty that function and method have the same name. We may leave it as is or could change the function name instead (which defined in `sc/utils`). `save_model` -> `save_output_model`. It will not interfere with the interface.

Reply:
Please, take a look.